tools/klee-ptree/main.cpp

```diff
@@ -0,0 +1,53 @@
+//===-- main.cpp ------------------------------------------------*- C++ -*-===//
+//
+//                     The KLEE Symbolic Virtual Machine
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cstdlib>
+#include <iostream>
+
+#include "Info.h"
+#include "Dot.h"
+
+
+void print_usage() {
+  std::cout << "Usage: klee-ptree <option> /path/ptree.db\n\n"
+               "Options:\n"
+               "\tbranchinfo  -  print branch statistics\n"
+               "\tdot         -  print tree in .dot format\n"
+               "\ttreeinfo    -  print tree statistics"
+               "\n";
+}
+
+
+int main(int argc, char * argv[]) {
+  if (argc != 3) {
+    print_usage();
+    exit(1);
+  }
+
+  // parse options
+  void (*action)(const Tree&);
+  std::string option(argv[1]);
```

Comment:
Why not use LLVM CL parsing?

Reply:
Then I have to link it and get that broken option syntax.

Reply:
Still better than rolling your own? And considering everything else gets linked with LLVM that's no excuse.

Reply:
Sure, when there is a use case for more complex flags in the future...

Reply:
I also agree that using the LLVM CL would make more sense, as it's what we use in other tools, such as kleaver.

Reply:
We _only_ use it for klee and kleaver but we don't use it in ktest-gen, ktest-randgen, klee-replay and of course the Python tools. It makes no sense to me to link against a fat library to parse a single string, I'd rather switch to getopt in that case. Esp. in my case, I'd like to analyse ptrees on a machine that doesn't even have a LLVM installation.